From 08c7e9ac12a9b3bb1a94486c3721b5698c9d7701 Mon Sep 17 00:00:00 2001
From: Samuel Ortiz <sameo@linux.intel.com>
Date: Thu, 13 Nov 2014 11:49:46 +0100
Subject: [PATCH 2/2] soc: sst: Memory leak fix with !CONFIG_DEBUG_FS

The iram and dram buffer needs to be freed on the error path,
regardless of CONFIG_DEBUG_FS value.

Change-Id: I204323f48a1baad7a0244d4e2e3000716241df90
Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
---
 sound/soc/intel/sst/sst.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/sound/soc/intel/sst/sst.c b/sound/soc/intel/sst/sst.c
index 6073ec4..c112ff2 100644
--- a/sound/soc/intel/sst/sst.c
+++ b/sound/soc/intel/sst/sst.c
@@ -1002,13 +1002,11 @@ do_free_probe_bytes:
 	if (sst_drv_ctx->pci_id == SST_CLV_PCI_ID)
 		kfree(sst_drv_ctx->probe_bytes);
 do_free_dram_buf:
-#ifdef CONFIG_DEBUG_FS
 	if (sst_drv_ctx->pci_id == SST_CLV_PCI_ID)
 		kfree(sst_drv_ctx->dump_buf.dram_buf.buf);
 do_free_iram_buf:
 	if (sst_drv_ctx->pci_id == SST_CLV_PCI_ID)
 		kfree(sst_drv_ctx->dump_buf.iram_buf.buf);
-#endif
 do_unmap_dram:
 	iounmap(sst_drv_ctx->dram);
 do_unmap_iram:
-- 
1.9.1

